integration-cli/docker_api_containers_test.go

```diff
@@ -714,6 +716,85 @@ func (s *DockerSuite) TestContainerApiCreateWithHostName(c *check.C) {
 	}
 }
 
+func (s *DockerSuite) TestContainerApiCreateWithDomainName(c *check.C) {
+	var domainName = "test-domain"
+	config := map[string]interface{}{
+		"Image":      "busybox",
+		"Domainname": domainName,
+	}
+
+	_, b, err := sockRequest("POST", "/containers/create", config)
+	if err != nil && !strings.Contains(err.Error(), "200 OK: 201") {
+		c.Fatal(err)
+	}
+
+	var container types.ContainerCreateResponse
+	if err := json.Unmarshal(b, &container); err != nil {
+		c.Fatal(err)
+	}
+
+	var id = container.ID
+
+	_, bodyGet, err := sockRequest("GET", "/containers/"+id+"/json", nil)
+
+	type configLocal struct {
+		Domainname string
+	}
+	type getResponse struct {
+		Id     string
+		Config configLocal
+	}
+
+	var containerInfo getResponse
+	if err := json.Unmarshal(bodyGet, &containerInfo); err != nil {
+		c.Fatal(err)
+	}
+	var domainNameActual = containerInfo.Config.Domainname
+	if domainNameActual != domainName {
+		c.Fatalf("Mismatched Hostname, Expected %v, Actual: %v ", domainName, domainNameActual)
+	}
+}
+
+func (s *DockerSuite) TestContainerApiCreateNetworkMode(c *check.C) {
+	UtilCreateNetworkMode(c, "host")
+	UtilCreateNetworkMode(c, "bridge")
+	UtilCreateNetworkMode(c, "container:web1")
+}
+
+func UtilCreateNetworkMode(c *check.C, networkModeParam string) {
+	var networkMode = networkModeParam
```

Comment:
Its not a big deal if you don't want to change it, but why not just call the param "networkMode" instead of "networkModeParam", then you don't need this assignment at all?